packages/xod-client/src/project/components/nodeParts/JumperNodeBody.jsx

```diff
@@ -2,31 +2,27 @@ import R from 'ramda';
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import { foldEither } from 'xod-func-tools';
+import { foldEither, foldMaybe, maybePath } from 'xod-func-tools';
 import { NODE_CORNER_RADIUS } from '../../nodeLayout';
 
 const INPUT_PINKEY = '__in__';
 const OUTPUT_PINKEY = '__out__';
 
+const getDeducedTypeOfPin = R.curry((pinkey, pins) =>
+  R.compose(
+    foldMaybe('generic', R.identity),
+    R.map(foldEither('error', R.identity)),
+    maybePath([INPUT_PINKEY, 'deducedType'])
+  )(pins)
+);
+
 const JumperNodeBody = ({ pins }) => {
   const inConnected = R.path([INPUT_PINKEY, 'isConnected'], pins);
   const outConnected = R.path([OUTPUT_PINKEY, 'isConnected'], pins);
 
   const type = R.cond([
-    [
-      () => inConnected,
-      R.compose(
-        foldEither('error', R.identity),
-        R.path([INPUT_PINKEY, 'deducedType'])
-      ),
-    ],
-    [
-      () => outConnected,
-      R.compose(
-        foldEither('error', R.identity),
-        R.path([OUTPUT_PINKEY, 'deducedType'])
-      ),
-    ],
+    [() => inConnected, getDeducedTypeOfPin(INPUT_PINKEY)],
+    [() => outConnected, getDeducedTypeOfPin(OUTPUT_PINKEY)],
     [R.T, R.always('')],
```

Comment:
Should this default to `'generic'` too?

Reply:
It does not make sense, but I'll do it to avoid questions in the future :)